cryptostore/data/gd.py

```diff
@@ -0,0 +1,158 @@
+
+
+def _get_folder_in_parent(drive, path) -> Tuple[str, str]:
+    """
+    Retrieve folder ID from given name and parent folder name.
+    If not existing, it is created.
+
+    Parameters:
+        drive:
+            Service with which interacting with Google Drive.
+        path (str):
+            path = '{prefix}/{exchange}/{data_type}/{pair}/
+                        {exchange}-{data_type}-{pair}-{int(timestamp)}.parquet'
+            String from which is retrieved `prefix` (parent folder) and name of
+            child folder '{exchange}-{data_type}-{pair}'.
+
+    Returns:
+        folder_id, folder_name (str):
+            Id of child folder '{exchange}-{data_type}-{pair}'. Create it if
+            not existing.
+
+    """
+
+    # Retrieve parent folder (prefix), and child folder.
+    path_struct = path.split('/')
+    folder_name = '-'.join(path_struct[1:4])
+    if len(path_struct) > 5:
+        # If larger than 5, it means prefix is more than a single folder.
+        # This case is not supported.
+        raise InconsistentStorage("Prefix {!s} appears to be a path. Only a \
+single folder name is accepted.".format(folder_name))
+
+    parent_name = path_struct[0]
+    # Retrieve candidates for child and parent folders.
+    res = drive.files().list(q="(name = '" + parent_name + "' or name = '"
+                                           + folder_name + "') and mimeType = \
+'application/vnd.google-apps.folder' and trashed = false",
+                             pageSize=20,
+                             fields='files(id, name, parents)').execute()
+    folders = res.get('files', [])
+
+    # Manage parent folder.
+    p_folders = [(folder['id'], folder['name']) for folder in folders
+                 if folder['name'] == parent_name]
+    if len(p_folders) > 1:
+        # If more than 2 folders with the same name, throw an error. We do not
+        # know which one is the right one to record data.
+        raise InconsistentStorage("At least 2 parent folders identified with \
+name {!s}. Please, make sure to provide a prefix corresponding to a unique \
+folder name in your Google Drive space.".format(parent_name))
+    elif not p_folders:
+        # If parent folder is not found, ask the user to create one.
+        raise InconsistentStorage("No existing folder found with name {!s}. \
+Please, make sure to provide a prefix corresponding to an existing and \
+accessible folder.".format(parent_name))
+    else:
+        p_folder_id = p_folders[0][0]
+
+    # Manage child folder.
+    c_folders = [(folder['id'], folder['name']) for folder in folders
+                 if ((folder['name'] == folder_name) and ('parents' in folder)
+                     and (p_folder_id in folder['parents']))]
+    if len(c_folders) > 1:
+        # If more than 2 folders with the same name, throw an error. We do not
+        # know which one is the right one to record data.
+        raise InconsistentStorage("At least 2 folders identified with name \
```

Comment:
i prefer long lines to lines shortened/continued with \. Its 2020, people have 4k monitors and other large resolution displays